src/ivc/public_params.rs

```diff
@@ -0,0 +1,234 @@
+use std::{fmt, num::NonZeroUsize};
+
+use ff::{FromUniformBytes, PrimeFieldBits};
+use group::prime::PrimeCurveAffine;
+use halo2curves::CurveAffine;
+use serde::Serialize;
+
+use crate::{
+    commitment::CommitmentKey,
+    digest::{self, DigestToCurve},
+    plonk::PlonkStructure,
+    poseidon::ROPair,
+    table::TableData,
+};
+
+use super::step_circuit::SynthesizeStepParams;
+
+pub(crate) struct CircuitPublicParams<C, RP>
+where
+    C: CurveAffine,
+    C::Base: PrimeFieldBits + FromUniformBytes<64>,
+    RP: ROPair<C::Base>,
+{
+    pub ck: CommitmentKey<C>,
+    pub params: SynthesizeStepParams<C, RP::OnCircuit>,
+    pub td: TableData<C::Base>,
```

Comment:
since we serialize plonk_structure below. Is it possible replace td field by PlonkStructure in CircuiPublicParams?